test/functional/test_framework/test_wrapper.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014-2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Wrapper Class for BitcoinTestFramework.
+
+The TestWrapper class extends the BitcoinTestFramework
+rpc & daemon process management functionality to external
+python environments.
+
+It is a singleton class, which ensures that users only
+start a single TestWrapper at a time."""
+
+import argparse
+from os import getpid
+from os.path import abspath, join
+
+from test_framework.test_framework import BitcoinTestFramework
+
+class TestWrapper:
+
+    class __TestWrapper(BitcoinTestFramework):
+
+        def set_test_params(self):
+            pass
+
+        def run_test(self):
+            pass
+
+        def setup(self, **kwargs):
+
+            if self.running:
+                print("TestWrapper is already running!")
```

Comment:
Make any sense to log instead of print? (same question for below)

Reply:
Thanks @jamesob for the review. For my understanding, do you mean logging it to the (already) running TestShell/Driver instance?

Reply:
As of now, I left this as `print(()`.